echoregions/regions2d/regions2d.py

```diff
@@ -501,3 +501,226 @@ def mask(
             M = M.rename(mask_var)
 
         return M
+
+    def transect_mask(
+        self,
+        da_Sv: DataArray,
+        transect_dict: dict = {
+            "start": "ST",
+            "break": "BT",
+            "resume": "RT",
+            "end": "ET",
+        },
+        file_names: List[str] = None,
+    ) -> DataArray:
+        """Mask data from Data Array containing Sv data based off of a Regions2D object
+        and its transect_values.
+
+        Parameters
+        ----------
+        da_Sv : Data Array
+            DataArray of shape (ping_time, depth) containing Sv data.
+        transect_dict : dict
+            Dictionary for transect values. Values must be unique and must be of same length.
+        file_names : list
+            List for file names to do the transect querying on.
+        Returns
+        -------
+        M : Data Array
+            A DataArray masked by the transect values from the Regions2d.data dataframe.
+        """
+
+        # Get transect strings
+        start_str = transect_dict["start"]
+        break_str = transect_dict["break"]
+        resume_str = transect_dict["resume"]
+        end_str = transect_dict["end"]
+        transect_strs = [start_str, break_str, resume_str, end_str]
+
+        # Check that there are 4 unique transect strings
+        if len(transect_strs) != len(set(transect_strs)):
+            raise ValueError(
+                "There exist duplicate values in transect_dict. \
+                             All values must be unique"
+            )
+
+        # Check that transect strings are of same length
```

Comment:
Yes, that does refer to the number of characters in "ST". It was a shoddy solution to removing the `11` in the case where we have `ST11`. I think a better way to do it would be to just remove all string values in region_names that are not in the transect dictionary.